### PR TITLE
Manual engine roll for 2019-03-19

### PR DESCRIPTION
**Pull request by bkonyi**
Manual roll due to some changes in the compilation pipeline that required changes in both the engine and framework tools.

### CODE REVIEW

**Comment by bkonyi** [comment]
@aam would you mind linking to the PR with the patch here? It seems I've misplaced it.

**Comment by bkonyi** [comment]
Still waiting on artifacts to be built. Will kick the bots again once that finishes.

File: packages/flutter_tools/lib/src/compile.dart [hunk 1]
```diff
@@ -453,10 +481,13 @@ class ResidentCompiler {
         ? _mapFilename(request.mainPath, packageUriMapper) + ' '
         : '';
     _server.stdin.writeln('recompile $mainUri$inputKey');
+    printTrace('<- recompile $mainUri$inputKey');
     for (String fileUri in request.invalidatedFiles) {
       _server.stdin.writeln(_mapFileUri(fileUri, packageUriMapper));
+      printTrace('<- ${_mapFileUri(fileUri, packageUriMapper)}');
     }
     _server.stdin.writeln(inputKey);
+    printTrace('<- $inputKey');
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/compile.dart]: What are the trace statements for? These only show up in verbose logs

Comment (bkonyi) [reply packages/flutter_tools/lib/src/compile.dart]: @aam

Comment (aam) [reply packages/flutter_tools/lib/src/compile.dart]: They show interaction between frontend server and compiler client.

**Comment by aam** [comment]
https://github.com/flutter/flutter/pull/29004 is the PR with changes to compile.dart to accommodate list of files now returned by the frontend server.

**Comment by aam** [comment]
> @aam would you mind linking to the PR with the patch here? It seems I've misplaced it.

I think you want to pick up updated PR as it has additional changes to the tests.

File: packages/flutter_tools/lib/src/compile.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/compile.dart]: Even if logs are are hidden we're still spending time building the strings, quite a lot in the case of the initial reload

Comment (aam) [reply packages/flutter_tools/lib/src/compile.dart]: Do you know time spent here is noticeable? I found that diagnostic to be useful, which is why I added it.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/compile.dart]: There is absolutely no reason to leave debugging code in production here

**Review by aam — APPROVED** [review]

File: packages/flutter_tools/lib/src/compile.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/lib/src/compile.dart]: I thin we frequently use `printTrace` when running in verbose mode we want to get clarity about what is going with the system to help troubleshoot issues for end users. This is no different.

**Pull request merged**